### DISCUSSION TITLE
API Recording bitrate?

### DISCUSSION THREAD

**Discussion by mkipness** [ant-media/Ant-Media-Server#5491] [General]
Hello,

I have used the api start/stop for recording live streams to mp4. It works fine.

However, I need a way to configure the sizing. With the default, the size of the recordings is twice the size of the recordings from OBS's record feature. I think due to the bitrate settings on OBS record.

How can I change the bitrate at which the API records?

Thanks,
Max

**Comment by USAMAWIZARD** [comment]
you can use this approach.
follow [this documentation](https://antmedia.io/docs/guides/recording-live-streams/user-defined-scripts/#mp4-muxingrecording-finish-process) to define a script which will be called whenever a file recording process has been completed.the scirpt will be called with the argument of path of the mp4 file.

in the script file  you can write some thing like this 
set your desired bitrate value in the command.

```
ffmpeg -i $1 -vcodec h264 -b:v 700k -acodec mp3 ./outputfile_name.mp4

```

OR ELSE

```
ffmpeg -i $1 -vcodec libx264 -crf 24 ./outputfile_name.mp4

```
-crf [crf]: Specifies the Constant Rate Factor (CRF) for the H. 264 encoder. The CRF ranges from 0-51,
 with 0 being lossless and 51 being the worst quality possible.

**Reply by mkipness** [reply]
Thanks for the response, but I was already aware of this approach. The problem is the time it takes on larger files. If you record with the correct bitrate to begin with, there is no extra wait time. That is what I was looking for.

Max

**Reply by USAMAWIZARD** [reply]
you can add adaptive bitrate settings , but this will have high load on the server side.
![image](https://github.com/ant-media/Ant-Media-Server/assets/47350008/48a434a7-f944-469d-8e5c-1620aacc6776)
